From 00b5d547fee762e3e9d68b493473f291d774b347 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Fr=C3=A9d=C3=A9ric=20Pierret=20=28fepitre=29?=
 <frederic.pierret@qubes-os.org>
Date: Sat, 16 May 2020 11:44:28 +0200
Subject: [PATCH 4/8] Don't reset client struct

Do not reset client->*_timeout_id fields in clientFrame(). The client struct is
allocated with g_new0(), which fill it with 0 already. But at this point there
may be some data already - especially after Qubes frames patch calling
clientUpdateName() earlier.

Authors:
@marmarek

Original work done in:
- QubesOS/qubes-desktop-linux-xfce4
- QubesOS/qubes-desktop-linux-xfce4-xfwm4
---
 src/client.c | 11 -----------
 1 file changed, 11 deletions(-)

diff --git a/src/client.c b/src/client.c
index 83be4b288..efb90c361 100644
--- a/src/client.c
+++ b/src/client.c
@@ -1885,17 +1885,6 @@ clientFrame (DisplayInfo *display_info, Window w, gboolean recapture)
         FLAG_SET (c->xfwm_flags, XFWM_FLAG_OPACITY_LOCKED);
     }
 
-    /* Timout for asynchronous icon update */
-    c->icon_timeout_id = 0;
-    /* Timout for asynchronous frame update */
-    c->frame_timeout_id = 0;
-    /* Timeout for blinking on urgency */
-    c->blink_timeout_id = 0;
-    /* Ping timeout  */
-    c->ping_timeout_id = 0;
-    /* Ping timeout  */
-    c->ping_time = 0;
-
     c->class.res_name = NULL;
     c->class.res_class = NULL;
     XGetClassHint (display_info->dpy, w, &c->class);
-- 
2.48.1

